staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -169,7 +169,7 @@ func (f *structuredMergeManager) Apply(liveObj, patchObj runtime.Object, managed
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to convert new object to proper version: %v", err)
 	}
-	f.objectDefaulter.Default(newObjVersioned)
+	f.objectDefaulter.Default(newObjVersioned)  // This is where I expected defaulting to actually happen
```

Comment:
Yeah, I think there are two places where I would have expected defaults:
1. Before converting objects to various versions to check for conflicts,
2. Here, as we've changed the object, possibly re-defaulting fields that have been pruned